CoreScriptsRoot/Modules/Server/ClientChat/ObjectPool.lua

```diff
@@ -0,0 +1,54 @@
+--	// FileName: ObjectPool.lua
+--	// Written by: TheGamer101
+--	// Description: An object pool class used to avoid unnecessarily instantiating Instances.
+
```

Comment:
Just checking, how similar is this to the dev console objectpool?
And if we like this one better, is this generalized enough that we could use it elsewhere in the chat/corescript code?